services/skus/service.go

```diff
@@ -1861,22 +1840,26 @@ func (s *Service) createOrder(ctx context.Context, req *model.CreateOrderRequest
 				return nil, fmt.Errorf("failed to create checkout session: %w", err)
 			}
 
-			if err := s.Datastore.AppendOrderMetadata(ctx, &order.ID, "radomCheckoutSessionId", ssid); err != nil {
+			if err := s.orderRepo.AppendMetadata(ctx, tx2, order.ID, "radomCheckoutSessionId", ssid); err != nil {
 				return nil, fmt.Errorf("failed to update order metadata: %w", err)
 			}
 		}
 	}
 
-	if err := s.updateOrderIntvals(ctx, order.ID, items, numIntervals); err != nil {
+	if err := s.updateOrderIntvals(ctx, tx2, order.ID, items, numIntervals); err != nil {
 		return nil, fmt.Errorf("failed to update order metadata: %w", err)
 	}
 
+	if err := tx2.Commit(); err != nil {
+		return nil, err
+	}
+
 	return order, nil
 }
 
-func (s *Service) updateOrderIntvals(ctx context.Context, oid uuid.UUID, items []model.OrderItem, nintvals int) error {
+func (s *Service) updateOrderIntvals(ctx context.Context, dbi sqlx.ExecerContext, oid uuid.UUID, items []model.OrderItem, nintvals int) error {
```

Comment:
nit: the name `intvals` is a little ambiguous e.g could be mistaken as Int Values

We could rename to `updateOrderIntervals`

Reply:
Thanks, updated.